product_configurator/models/product.py

```diff
@@ -369,19 +369,25 @@ def value_available(self, attr_val_id, value_ids):
             lambda l: attr_val_id in l.value_ids.ids
         )
 
-        domains = config_lines.mapped('domain_id').compute_domain()
-
-        for domain in domains:
-            if domain[1] == 'in':
-                if not set(domain[2]) & set(value_ids):
-                    return False
-            else:
-                if set(domain[2]) & set(value_ids):
-                    return False
+        for config_line in config_lines:
+            domains = config_line.mapped('domain_id').compute_domain()
+            for domain in domains:
+                if domain[1] == 'in':
+                    if not set(domain[2]) & set(value_ids):
+                        # Don't break if the module has not been upgraded...
```

Comment:
Could you add a TODO here to remove the check for field existence in future versions?